core-web/apps/dotcms-ui/src/app/portlets/dot-containers/dot-container-create/dot-container-code/dot-container-code.component.spec.ts

```diff
@@ -0,0 +1,190 @@
+import { CoreWebService, CoreWebServiceMock } from '@dotcms/dotcms-js';
+import { DotCMSContentType } from '@dotcms/dotcms-models';
+import { ButtonModule } from 'primeng/button';
+import { DialogService, DynamicDialogModule } from 'primeng/dynamicdialog';
+import { Menu, MenuModule } from 'primeng/menu';
+import { TabViewModule } from 'primeng/tabview';
+import { of } from 'rxjs';
+import { DotAddVariableModule } from './dot-add-variable/dot-add-variable.module';
+import { DotContentEditorComponent } from './dot-container-code.component';
+
+const mockContentTypes: DotCMSContentType[] = [
+    {
+        baseType: 'CONTENT',
+        clazz: 'com.dotcms.contenttype.model.type.ImmutableSimpleContentType',
+        defaultType: false,
+        description: 'Activities available at desitnations',
+        detailPage: 'e5f131d2-1952-4596-bbbf-28fb28021b68',
+        fixed: false,
+        folder: 'SYSTEM_FOLDER',
+        host: '48190c8c-42c4-46af-8d1a-0cd5db894797',
+        iDate: 1567778770000,
+        icon: 'paragliding',
+        id: '778f3246-9b11-4a2a-a101-e7fdf111bdad',
+        modDate: 1663219138000,
+        multilingualable: false,
+        nEntries: 10,
+        name: 'Activity',
+        system: false,
+        urlMapPattern: '/activities/{urlTitle}',
+        variable: 'Activity',
+        versionable: true,
+        workflows: [],
+        fields: [],
+        layout: []
+    }
+];
+
+@Component({
+    selector: 'dot-host-component',
+    template: `
+        <dot-container-code [contentTypes]="contentTypes" [fg]="form"> </dot-container-code>
+    `
+})
+class HostTestComponent {
+    form: FormGroup;
+    contentTypes = mockContentTypes;
+    constructor(private fb: FormBuilder) {
+        this.form = this.fb.group({
+            containerStructures: this.fb.array([])
+        });
+
+        (this.form.get('containerStructures') as FormArray).push(
+            this.fb.group({
+                structureId: new FormControl('12345'),
+                code: new FormControl('')
+            })
+        );
+    }
+}
+
+@Component({
+    selector: 'dot-textarea-content',
+    template: '',
+    providers: [
+        {
+            multi: true,
+            provide: NG_VALUE_ACCESSOR,
+            useExisting: forwardRef(() => DotTextareaContentMockComponent)
+        }
+    ]
+})
+export class DotTextareaContentMockComponent implements ControlValueAccessor {
+    @Input()
+    code;
+
+    @Input()
+    height;
+
+    @Input()
+    show;
+
+    @Input()
+    value;
+
+    @Input()
+    width;
+
+    @Output()
+    monacoInit = new EventEmitter();
+
+    @Input()
+    language;
+
+    writeValue() {
+        //
+    }
+    registerOnChange() {
+        //
+    }
+    registerOnTouched() {
+        //
+    }
+}
+
+const messageServiceMock = new MockDotMessageService({
+    'containers.properties.add.variable.title': 'Title',
+    Add: 'Add'
+});
+
+describe('DotContentEditorComponent', () => {
```

Comment:
You are not testing the core functionality of this component here.

You need to tests all the add and remove content types.

Reply:
Updated